File: src/gestorAplicacion/tienda/Servicio.java
package gestorAplicacion.tienda;
import java.io.Serializable;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

import gestorAplicacion.personal.Dependiente;
import gestorAplicacion.personal.Tecnico;
/**
 * 
 * @author Felipe Miranda
 * @summary Servicio deberia contener toda la informacion que relaciona a un cliente y su producto a reparar 
 * con los empleados (el dependiente que le atendio y el tecnico encargado de la reparacion) del sistema. 
 *
 */
public class Servicio implements Serializable {
	private static final long serialVersionUID = 1L;
	private int identificador;

	private static List<Servicio> servicios;
	private Tecnico tecnico;
	private boolean pagado;
	private Producto producto;
	private Date fecha;
	private Cliente cliente;
	private Dependiente dependiente;
	private double costo;
	private String diagnostico;
	private boolean reparado;
	
	static {
		servicios = new ArrayList<Servicio>();
	}
		

	public Servicio(Tecnico tecnico, Producto producto, Cliente cliente,
			Dependiente dependiente) {
		this.tecnico = tecnico;
		this.producto = producto;
		this.setFecha(new Date());
		this.cliente = cliente;
		this.dependiente = dependiente;
		this.reparado = false;
		if (servicios.isEmpty())
			identificador = 0;
		else 
			identificador = servicios.size();
		
		servicios.add(this);
	}


	public void setPagado(boolean pagado){
		this.pagado = pagado;
	}
	
	
	public Producto getProducto() {
		return producto;
	}


	public Dependiente getDependiente() {
		return dependiente;
	}
	/**
	 * 
	 * @summary costo seria la suma de los diferentes precios de cada componente que se arregle. 
	 * Si se va a agregar una pieza a la reparacion del producto, se debe sumar aca su total. 
	 */
	public void anadirCosto(double precio) {
		this.costo+=precio;
	}
	
	public double getCosto() {
		return costo;
	}

	public String getDiagnostico() {
		return diagnostico;
	}

	public void setDiagnostico(String diagnostico) {
		this.diagnostico = diagnostico;
	}	
	
	public boolean isPagado() {
		return pagado;
	}
	
	public void setCosto(double costo) {
		this.costo = costo;
	}

	public int getIdentificador() {
		return identificador;
	}

	public Cliente getCliente() {
		return cliente;
	}

	public void setCliente(Cliente cliente) {
		this.cliente = cliente;
	}

	public static List<Servicio> getServicios() {
		return servicios;
	}

	public static void setServicios(List<Servicio> servicios) {
		Servicio.servicios = servicios;
	}

	public Tecnico getTecnico() {
		return tecnico;
	}

	public Date getFecha() {
		return fecha;
	}

	public void setFecha(Date fecha) {
		this.fecha = fecha;
	}

	public boolean isReparado() {
		return reparado;
	}

	public void setReparado(boolean reparado) {
		this.reparado = reparado;
	}
	
	public String toString() {
		return "\nIdentificador del servicio: " + this.identificador
				+ "\nCliente: " + this.cliente
				+ "\nProducto asociado: " + this.producto
				+ "\nReparado: " + this.reparado
				+ "\nPagado: " + this.pagado + "\n";
	}
	
	
	
}

File: practica2/generic_it/gestor_aplicacion/tienda/servicio.py


from queue import Empty


class servicio:
    servicios = list()
    def __init__(self, _tecnico, _producto, _cliente, _dependiente):
        self._tecnico = _tecnico
        self._producto = _producto
        self._cliente = _cliente
        self._dependiente = _dependiente
        self._reparado = False
        self._costo = 0
        self._diagnostico = ""
        if self.servicios is Empty:
            self.identificador = 0
        else:
            self.identificador = self.servicios.size
        self.servicios.append(self)

    def setPagado(self, _pagado):
        self._pagado = _pagado

    def getProducto(self):
        return self._producto

    def getDependiente(self):
        return self._dependiente

    def anadirCosto(self, precio):
        self.costo+=precio

    def getCosto(self):
        return self._costo

    def getDiagnostico(self):
        return self._diagnostico

    def isPagado(self):
        return self._pagado

    def setCosto(self, costo):
        self._costo=costo

    def getIdentificador(self):
        return self.identificador

    def getCliente(self):
        return self._cliente

    def setCliente(self, cliente):
        self._cliente = cliente

    @classmethod
    def getServicios(cls):
        return cls.servicios

    @classmethod
    def setServicios(cls, servicios):
        cls.servicios = servicios

    def getTecnico(self):
        return self._tecnico

    def isReparado(self):
        return self._reparado

    def setReparado(self, reparado):
        self._reparado = reparado

    def toString(self):
        return "\nIdentificador del servicio: " + self.identificador+ "\nCliente: " + self._cliente+ "\nProducto asociado: " + self._producto+ "\nReparado: " + self._reparado+ "\nPagado: " + self._pagado + "\n"